build/xtask/src/clippy.rs

```diff
@@ -2,74 +2,86 @@
-            .and_then(|v| v.as_str())
-            .expect("Couldn't find [package.name]; pass -p <package> to run clippy on a specific package or --all to check all packages")
-            .to_string()
-    });
+    let toml = Config::from_file(&cfg)?;
+    let task_names = toml.tasks.keys().cloned().collect::<Vec<_>>().join(",");
+    let shared_syms: Option<&[String]> = None;
 
-    println!("Running clippy on: {}", package);
+    let mut src_dir = cfg.to_path_buf();
+    src_dir.pop();
+    let src_dir = src_dir;
 
-    let mut cmd = Command::new("cargo");
-    cmd.arg("clippy");
-    // TODO: Remove this argument once resolved on stable:
-    //
-    // https://github.com/rust-lang/rust-clippy/issues/4612
-    //
-    // TL;DR: Switching back and forth between "clippy" and "check"
-    // caches the results from one execution. This means a succeeding
-    // "check" execution may hide a failing "clippy" execution.
-    cmd.arg("-Zunstable-options");
-    cmd.arg("-p");
-    cmd.arg(&package);
+    if tasks.is_empty() {
+        bail!("Must provide one or more task names");
+    }
 
-    if let Some(target) = target {
-        cmd.arg("--target");
-        cmd.arg(target);
+    for name in &tasks {
+        if !toml.tasks.contains_key(name) {
+            bail!("{}", toml.task_name_suggestion(name));
+        }
     }
 
-    // Clippy arguments
-    cmd.arg("--");
-    cmd.arg("-D");
-    cmd.arg("clippy::all");
-    cmd.arg("-A");
-    cmd.arg("clippy::missing_safety_doc");
-    cmd.arg("-A");
-    cmd.arg("clippy::identity_op");
-    cmd.arg("-A");
-    cmd.arg("clippy::too_many_arguments");
+    for (i, name) in tasks.iter().enumerate() {
+        let task_toml = &toml.tasks[name];
+        if tasks.len() > 1 {
+            if i > 0 {
+                println!();
+            }
+            println!(
+                "================== {} [{}] ==================",
+                name, task_toml.name
+            );
+        }
 
-    for opt in options {
-        cmd.arg(opt);
-    }
+        let build_config = BuildConfig::new(
+            &cfg,
+            &toml.target,
+            &toml.board,
+            verbose,
+            Some(&task_toml.name),
+            &task_toml.features,
+            &task_names,
+            &toml.secure_separation,
+            &shared_syms,
+            &task_toml.config,
+            &toml.config,
+            &[],
+        );
+        let mut cmd = Command::new("cargo");
+        cmd.arg("clippy");
+        build_config.apply(&mut cmd);
 
-    // this is only actually used for demo-stm32h7 but is harmless to include, so let's do
-    // it unconditionally
-    cmd.env("HUBRIS_BOARD", "nucleo-h743zi2");
+        cmd.arg("--");
+        cmd.arg("-W");
+        cmd.arg("clippy::all");
+        cmd.arg("-A");
+        cmd.arg("clippy::missing_safety_doc");
+        cmd.arg("-A");
+        cmd.arg("clippy::identity_op");
+        cmd.arg("-A");
+        cmd.arg("clippy::too_many_arguments");
```

Comment:
This was how it looked when I got here, so I decided not to change it 😆 

(other than changing `-D` to `-W`, because otherwise it hit [this line](https://github.com/oxidecomputer/hubris/blob/master/sys/abi/src/lib.rs#L14) and didn't check anything else)

Reply:
😅 wonder how that happened... seems fine then